src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -381,4 +382,41 @@ private int getClientCount(final String clientList) {
     return clientList.split("\n").length;
   }
 
+  @Test(expected = JedisConnectionException.class)
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000, "foobared", 0, "my_shiny_client_name", false, null, null, null);
+    JedisPool pool = new JedisPool(new JedisPoolConfig(), factory);
+    Jedis obj1 = pool.getResource();
+    obj1.set("foo", "bar");
+    assertEquals("bar", obj1.get("foo"));
+    assertEquals(1, pool.getNumActive());
```

Comment:
@HeartSaVioR @marcosnils Should we use it?

@yapei123 Wait for what others say.